percona-packaging/scripts/psmdb_builder.sh

```diff
@@ -0,0 +1,796 @@
+        sed -i 's|CXX = g++-5|CXX = /usr/bin/g++-7|' debian/rules
+        sed -i 's|CC = /usr/local/gcc-5.4.0/bin/gcc-5.4|CC = /usr/bin/gcc-7|' debian/rules
+        sed -i 's|CXX = /usr/local/gcc-5.4.0/bin/g++-5.4|CXX = /usr/bin/g++-7|' debian/rules
+        sed -i 's:release:release --disable-warnings-as-errors :g' debian/rules 
+    fi
+    if [ x"${DEBIAN}" = xbionic ]; then
+        sed -i "s/-Werror//g" src/third_party/PerconaFT/cmake_modules/TokuSetupCompiler.cmake
+    fi
+}
+
+install_deps() {
+    if [ $INSTALL = 0 ]
+    then
+        echo "Dependencies will not be installed"
+        return;
+    fi
+    if [ ! $( id -u ) -eq 0 ]
+    then
+        echo "It is not possible to instal dependencies. Please run as root"
+        exit 1
+    fi
+    CURPLACE=$(pwd)
+
+    if [ "x$OS" = "xrpm" ]; then
+      yum -y install wget
+      add_percona_yum_repo
+      wget http://jenkins.percona.com/yum-repo/percona-dev.repo
+      mv -f percona-dev.repo /etc/yum.repos.d/
+      yum clean all
+      yum -y install epel-release
+      rm -fr /usr/local/gcc-5.4.0
+      RHEL=$(rpm --eval %rhel)
+      if [ x"$RHEL" = x6 ]; then
+        yum -y install rpmbuild rpm-build libpcap-devel gcc make cmake gcc-c++ openssl-devel cyrus-sasl-devel snappy-devel zlib-devel bzip2-devel libpcap-devel scons make rpm-build rpmbuild percona-devtoolset-gcc percona-devtoolset-binutils percona-devtoolset-gcc-c++ percona-devtoolset-libstdc++-devel percona-devtoolset-valgrind-devel python27 python27-devel
+        wget https://bootstrap.pypa.io/get-pip.py
+        python2.7 get-pip.py
+        rm -rf /usr/bin/python2
+        ln -s /usr/bin/python2.7 /usr/bin/python2
+      else
+        yum -y install rpmbuild rpm-build libpcap-devel gcc make cmake gcc-c++ openssl-devel cyrus-sasl-devel snappy-devel zlib-devel bzip2-devel scons rpmlint rpm-build git python-pip python-devel libopcodes
+      fi
+      install_golang
+      install_gcc_54_centos
+    else
+      export DEBIAN=$(lsb_release -sc)
+      export ARCH=$(echo $(uname -m) | sed -e 's:i686:i386:g')
+      INSTALL_LIST="python python-dev valgrind scons liblz4-dev devscripts debhelper debconf libpcap-dev libbz2-dev libsnappy-dev pkg-config zlib1g-dev libzlcore-dev dh-systemd libsasl2-dev gcc g++ cmake "
+      if [ x"${DEBIAN}" = xstretch -o x"${DEBIAN}" = xbionic -o x"${DEBIAN}" = xartful ]; then
+        INSTALL_LIST="${INSTALL_LIST} libssl1.0-dev"
+      else
+        INSTALL_LIST="${INSTALL_LIST} libssl-dev"
+      fi
+      until apt-get -y install dirmngr; do
+        sleep 1
+        echo "waiting"
+      done
+      add_percona_apt_repo
+      until apt-get update; do
+        sleep 1
+        echo "waiting"
+      done
+      until apt-get -y install ${INSTALL_LIST}; do
+        sleep 1
+        echo "waiting"
+      done
+      install_golang
+      install_gcc_54_deb
+      wget https://bootstrap.pypa.io/get-pip.py
+      python get-pip.py
+      easy_install pip
```

Comment:
Isn't this installing pip twice - first python get-pip.py then easy_install pip?

Reply:
No. only during  `easy_install pip` it will be installed

Reply:
```
vagrant@t-ubuntu1604-64:~$ sudo python get-pip.py
The directory '/home/vagrant/.cache/pip/http' or its parent directory is not owned by the current user and the cache has been disabled. Please check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
The directory '/home/vagrant/.cache/pip' or its parent directory is not owned by the current user and caching wheels has been disabled. check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
Collecting pip
  Downloading https://files.pythonhosted.org/packages/0f/74/ecd13431bcc456ed390b44c8a6e917c1820365cbebcb6a8974d1cd045ab4/pip-10.0.1-py2.py3-none-any.whl (1.3MB)
    100% |████████████████████████████████| 1.3MB 3.0MB/s
Collecting setuptools
  Downloading https://files.pythonhosted.org/packages/8c/10/79282747f9169f21c053c562a0baa21815a8c7879be97abd930dbcf862e8/setuptools-39.1.0-py2.py3-none-any.whl (566kB)
    100% |████████████████████████████████| 573kB 9.0MB/s
Collecting wheel
  Downloading https://files.pythonhosted.org/packages/1b/d2/22cde5ea9af055f81814f9f2545f5ed8a053eb749c08d186b369959189a8/wheel-0.31.0-py2.py3-none-any.whl (41kB)
    100% |████████████████████████████████| 51kB 5.1MB/s
Installing collected packages: pip, setuptools, wheel
Successfully installed pip-10.0.1 setuptools-39.1.0 wheel-0.31.0

vagrant@t-ubuntu1604-64:~$ pip --version
pip 10.0.1 from /usr/local/lib/python2.7/dist-packages/pip (python 2.7)

vagrant@t-ubuntu1604-64:~$ pip list
Package    Version
---------- -------
pip        10.0.1
setuptools 39.1.0
wheel      0.31.0

vagrant@t-ubuntu1604-64:~$ sudo easy_install pip
Searching for pip
Best match: pip 10.0.1
Adding pip 10.0.1 to easy-install.pth file
Installing pip script to /usr/local/bin
Installing pip3.6 script to /usr/local/bin
Installing pip3 script to /usr/local/bin

Using /usr/local/lib/python2.7/dist-packages
Processing dependencies for pip
Finished processing dependencies for pip
```
From what I know pip and easy_install do the same/similar thing. But I'll leave it to you as I might be missing something here and I don't see that it hurts.